mm2src/coins/eth/erc20.rs

```diff
@@ -90,16 +91,22 @@ pub fn get_erc20_ticker_by_contract_address(ctx: &MmArc, platform: &str, contrac
     })
 }
 
-/// Finds an enabled ERC20 token by its contract address and returns it as `MmCoinEnum`.
-pub async fn get_enabled_erc20_by_contract(
+/// Finds an enabled ERC20 token by its chain id and contract address and returns it as `MmCoinEnum`.
+pub async fn get_enabled_erc20_by_contract_chain_id(
     ctx: &MmArc,
     contract_address: Address,
+    token_conf: &Json,
 ) -> MmResult<Option<MmCoinEnum>, String> {
+    let chain_id = token_conf["chain_id"]
```

Comment:
nit: we better let the param be the chain_id instead. that aligns more with the function name.